modules/rosa-hcp-byo-oidc.adoc

```diff
@@ -0,0 +1,38 @@
+// Module included in the following assemblies:
+//
+// * rosa_install_access_delete_clusters/rosa-sts-creating-a-cluster-quickly.adoc
+// * rosa_getting_started/quickstart.adoc
+
+:_content-type: PROCEDURE
+[id="rosa-hcp-byo-oidc_{context}"]
+= Configuring your own OpenID Connect provider
+
+You can use your own Open ID Connect (OIDC) provider with {product title} (ROSA) {hcp-first}.
+
+.Prequisites
+
+* You have a configured OIDC provider URL.
+
+.Procedure
+
+. In your terminal, run the following command to create your OIDC configuration alongside the AWS resources.
++
+[source,terminal]
+----
+$ rosa create oidc-config --mode auto
+----
++
+This command returns the following information.
++
+.Sample output
++
+[source,terminal]
+----
+I: This command will create a S3 bucket populating it with documents to be compliant with OIDC protocol. It will also create a Secret in Secrets Manager containing the private key.
+I: Please run command below to create a cluster with this oidc config:
+rosa create cluster --sts \
+ --oidc-endpoint-url https://oidc-l8c0.s3.us-east-1.amazonaws.com \
+ --oidc-private-key-secret-arn arn:aws:secretsmanager:us-east-1:269733383066:secret:rosa-private-key-oidc-l8c0-4vdMVv
+----
+
+You need to supply the URL for your S3 bucket in the OIDC Endpoint URL field. The CLI retrieves the Secrets Manager ARN. You can create your cluster using the command.
```

Comment:
Revised to:

> When creating your cluster, you must supply the OIDC endpoint URL and secret manager secret ARN. The CLI output provides both values for `--mode auto`, otherwise you must to determine these values based on `aws` CLI output for `--mode manual`.

But, I'm not sure I understand the second sentence. If I use `rosa create oidc-config` I should receive the private key and OIDC URL, which I did. When I entered `rosa create oidc-config --mode manual`, the CLI returned a bunch of prompts. Do I need to execute all of these as they are?

I just want to make sure I'm giving enough directions on this piece.